rpcserver/router.go

```diff
@@ -687,6 +729,209 @@ func (server *routedBoltzServer) CreateReverseSwap(ctx context.Context, request
+		}
+	}
+
+	fromName := request.GetFromWallet()
+	fromWallet, err := server.onchain.GetAnyWallet(onchain.WalletChecker{
+		Name:     fromName,
+		Currency: pair.From,
+		EntityId: macaroons.EntityFromContext(ctx),
+	})
+	externalPay := request.GetExternalPay()
+	if err != nil && !externalPay {
+		// the specified name was wrong
+		if fromName != "" {
+			return nil, handleError(err)
+		}
+		// no name was specified, which means there is no wallet available
+		externalPay = true
+	}
+
+	toName := request.GetToWallet()
+	toWallet, err := server.onchain.GetAnyWallet(onchain.WalletChecker{
+		Name:     request.GetToWallet(),
+		Currency: pair.To,
+		EntityId: macaroons.EntityFromContext(ctx),
+	})
+	if err != nil && toName != "" {
+		return nil, handleError(err)
+	}
+
+	response, err := server.boltz.CreateChainSwap(createChainSwap)
+
+	if err != nil {
+		return nil, handleError(errors.New("boltz error: " + err.Error()))
+	}
+
+	chainSwap := database.ChainSwap{
+		Id:                response.Id,
+		Pair:              pair,
+		State:             boltzrpc.SwapState_PENDING,
+		Error:             "",
+		Preimage:          preimage,
+		IsAuto:            isAuto,
+		ServiceFeePercent: utils.Percentage(chainPair.Fees.Percentage),
+	}
+
+	parseDetails := func(details *boltz.ChainSwapData, currency boltz.Currency) (*database.ChainSwapData, error) {
+		swapData := &database.ChainSwapData{
+			Id:                 response.Id,
+			Currency:           currency,
+			Amount:             details.Amount,
+			TimeoutBlockHeight: details.TimeoutBlockHeight,
+			Tree:               details.SwapTree.Deserialize(),
+			LockupAddress:      details.LockupAddress,
+		}
+		if currency == pair.From {
+			swapData.PrivateKey = refundPrivateKey
+			swapData.Address = request.GetRefundAddress()
+		} else {
+			swapData.PrivateKey = claimPrivateKey
+			swapData.Address = request.GetClaimAddress()
+		}
+
+		if swapData.Address != "" {
+			if err := boltz.ValidateAddress(server.network, swapData.Address, currency); err != nil {
+				return nil, err
+			}
+		}
+
+		swapData.TheirPublicKey, err = btcec.ParsePubKey(details.ServerPublicKey)
+		if err != nil {
+			return nil, err
+		}
+
+		if currency == boltz.CurrencyLiquid {
+			swapData.BlindingKey, _ = btcec.PrivKeyFromBytes(details.BlindingKey)
+		}
+
+		if err := swapData.InitTree(currency == pair.To); err != nil {
+			return nil, err
+		}
+
+		if err := swapData.Tree.Check(boltz.ChainSwap, swapData.TimeoutBlockHeight, preimageHash); err != nil {
+			return nil, err
+		}
+
+		if err := swapData.Tree.CheckAddress(details.LockupAddress, server.network, swapData.BlindingPubKey()); err != nil {
+			return nil, err
+		}
+
+		return swapData, nil
+	}
+
+	chainSwap.ToData, err = parseDetails(response.ClaimDetails, pair.To)
+	if err != nil {
+		return nil, handleError(err)
+	}
+	if toWallet != nil {
+		id := toWallet.GetWalletInfo().Id
+		chainSwap.ToData.WalletId = &id
```

Comment:
Maybe I missed it but where do we set the claim address if one was specified in the request?

Reply:
Here: https://github.com/BoltzExchange/boltz-client/pull/133/files#diff-3c0751abe8a29668b3f795bfe68a1a0dd18a58e1b76816afb7e4ce0d79cc28bcR836